dashboard/src/shared/Chart.ts

```diff
@@ -4,49 +4,40 @@ import { IChart, IChartVersion } from "./types";
 import * as URL from "./url";
 
 export default class Chart {
-  public static async fetchCharts(repo: string) {
-    const { data } = await axiosWithAuth.get<{ data: IChart[] }>(URL.api.charts.list(repo));
+  public static async fetchCharts(namespace: string, repo: string) {
+    const { data } = await axiosWithAuth.get<{ data: IChart[] }>(URL.api.charts.list(namespace, repo));
     return data.data;
   }
 
-  public static async fetchChartVersions(id: string) {
+  public static async fetchChartVersions(namespace: string, id: string) {
     const { data } = await axiosWithAuth.get<{ data: IChartVersion[] }>(
-      URL.api.charts.listVersions(id),
+      URL.api.charts.listVersions(namespace, id),
     );
     return data.data;
   }
 
-  public static async getChartVersion(id: string, version: string) {
+  public static async getChartVersion(namespace: string, id: string, version: string) {
     const { data } = await axiosWithAuth.get<{ data: IChartVersion }>(
-      URL.api.charts.getVersion(id, version),
+      URL.api.charts.getVersion(namespace, id, version),
     );
     return data.data;
   }
 
-  public static async getReadme(id: string, version: string) {
-    const { data } = await axiosWithAuth.get<string>(URL.api.charts.getReadme(id, version));
+  public static async getReadme(namespace: string, id: string, version: string) {
+    const { data } = await axiosWithAuth.get<string>(URL.api.charts.getReadme(namespace, id, version));
     return data;
   }
 
-  public static async getValues(id: string, version: string) {
-    const { data } = await axiosWithAuth.get<string>(URL.api.charts.getValues(id, version));
+  public static async getValues(namespace: string, id: string, version: string) {
+    const { data } = await axiosWithAuth.get<string>(URL.api.charts.getValues(namespace, id, version));
     return data;
   }
 
-  public static async getSchema(id: string, version: string) {
-    const { data } = await axiosWithAuth.get<JSONSchema4>(URL.api.charts.getSchema(id, version));
+  public static async getSchema(namespace: string, id: string, version: string) {
+    const { data } = await axiosWithAuth.get<JSONSchema4>(URL.api.charts.getSchema(namespace, id, version));
     return data;
   }
 
-  public static async exists(id: string, version: string, repo: string) {
-    try {
-      await axiosWithAuth.get(URL.api.charts.getVersion(`${repo}/${id}`, version));
-    } catch (e) {
-      return false;
-    }
-    return true;
-  }
-
   public static async listWithFilters(name: string, version: string, appVersion: string) {
```

Comment:
this function also requires to get the namespace, no?

Reply:
Indeed, updated, thanks.